docs/tutorials/understanding-the-rest-api.rst

```diff
@@ -0,0 +1,119 @@
+            record_serializers={
+                'application/json': ('invenio_records_rest.serializers'
+                                     ':json_v1_response'),
+            },
+            search_serializers={
+                'application/json': ('invenio_records_rest.serializers'
+                                     ':json_v1_search'),
+            },
+            record_loaders={
+                'application/json': ('invenio_records_rest.loaders'
+                                     ':json_pid_checker')
+            }
+        )
+    )
+
+In fact you can use the Invenio-Records-REST `example app <https://github.com/inveniosoftware/invenio-records-rest/tree/master/examples/>`_. You would just need to add to the configuration:
+
+.. code-block:: python
+
+    app.config['RECORDS_REST_ENDPOINTS']['recid']['record_loaders'] = {
+        'application/json': ('invenio_records_rest.loaders'
+                             ':json_pid_checker')
+    }
+
+
+Creating a record
+-----------------
+
+In the following flowchart, the process that a record goes through upon creation is shown:
+
+.. image:: ../_static/diagrams/create_record.png
+    :align: center
+    :scale: 80%
+
+`Note: If it is not displayed correctly click on it to be able to zoom in.`
+
+The record is sent by performing a POST operation, which hits the API through the `records` endpoint(`RecordsListResource: POST <https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L557>`_). For example:
+
+.. code-block:: shell
+
+    curl -k --header "Content-Type: application/json" \
+        --request POST \
+        --data '{"title":"Some title", "contributors": [{"name": "Doe, John"}]}' \
+        https://localhost:5000/api/records/?prettyprint=1
+
+We will now explain six steps, that correspond to the six big boxes of the previous diagram.
+
+The first step is to load the record, which as its name implies, is performed by the **loader**. At the beginning of this section the `json_pid_checker` was configured as the `record_loader` for `application/json` requests, therefore, it is the one used in this case.
+
+The record will be parsed by the `RecordMetadataSchemaJSONV1`. This schema only checks that the PID value is injected if needed (mostly for PUT operations). Nonetheless, you can create your own `Marshmallow` schema to fit your needs.
+
+The next step is to check if the user has permissions to create records, by default all authenticated users are allowed. You can see more about access control in the `managing access section </docs/tutorial-module/managing-access/index>`_.
+
+Aftewards, the record is minted. This action again as its name implies, is carried out by the **minter**. It adds to the record the corresponding value to the attribute configured via `PIDSTORE_RECID_FIELD <https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/config.py#L11>`_. For example, increasing by one its value.
```

Comment:
Generally avoid linking to GitHub repository code, since e.g. now the line numbers have changed. It's better to link to classes in docs or use `intersphinx`